exonum-java-binding-common/src/main/java/com/exonum/binding/common/configuration/ValidatorKey.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2018 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.common.configuration;
+
+import com.exonum.binding.common.hash.HashCode;
+import com.google.auto.value.AutoValue;
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * Public keys of validator nodes.
+ */
+@AutoValue
+public abstract class ValidatorKey {
+  /**
+   * Consensus key is used for messages related to the consensus algorithm.
+   */
+  @SerializedName("consensus_key")
+  public abstract HashCode consensusKey();
```

Comment:
Wait, it is not a HashCode, it is a PublicKey (same below).

Reply:
That will require a new JSON type adapter for PublicKey (maybe it is possible to make a single for AbstractKey that will work for both PublicKey & PrivateKey).

Reply:
Switching to PublicKey is simple, but i have doubts about single AbstractKeyGsonSerializer
How we should distinguish same public and private keys in Json representation and deserialize them to appropriate classes ? 
```
{
"publicKey" : "hexString",
"privateKey" : "hexString"
}
```